### ISSUE TITLE
Emissivity profiles

### ISSUE THREAD

**Issue by phajy** [phajy/OpticallyThinDiscs.jl#3]
The idea is to self-consistently calculate the illumination of the accretion disc from a particular corona model. We can start with a simple lamp post model and optically thick, geometrically thin accretion disk ($\theta = \pi/2$). This has been done extensively for the Kerr spacetime but we can generalise it for arbitrary spacetimes. We can then look at the line profiles produced by discs in a given spacetime that extend in to the innermost stable circular orbit allowed by that spacetime. It would be interesting to generalise this to thicker discs.

It will be interesting to see what other authors use for their emissivity profiles. A good starting point in the literature might be [Bambi et al. 2021](https://ui.adsabs.harvard.edu/abs/2021SSRv..217...65B). We will have to look at specific papers in detail to see how they calculate the emissivity profile in their model or whether they use a simple broken power law emissivity. There will be some degeneracy between emissivity profile, corona geometry, and spacetime properties.

Another good comparison paper for Kerr spacetimes is [Gonzalez et al. (2017)](https://ui.adsabs.harvard.edu/abs/2017MNRAS.472.1932G/abstract) as well as papers that reference / are referenced by this.

Later we could also consider the impact of returning radiation on the emissivity profile (and time delays).

**Comment by fjebaker** [comment]
Perhaps related to this is https://github.com/astro-group-bristol/Gradus.jl/issues/124, i.e. that the beaming effects of the source velocity have not been adequately checked / might be incorrect. I think there's a Wilkins & Fabian paper that loosely explores this, but doesn't satisfactorily state how they set the velocity of the source, and how they then used that in the transformation from local to global coordinates.

It would open up another parameter space to explore, which, also, might have variations in modified spacetimes -- who knows 🤷

**Comment by phajy** [comment]
Yes! Relativistic beaming is introduced in Wilkins's papers and we could try to check that too. I think they assume the emissivity of the corona is uniform. However, it would be good to self-consistently calculate the corona emissivity as well based on the thermal emission from the disc. But that might not be so quick and easy to implement (see, e.g., the work of [Ursini et al. 2020](https://ui.adsabs.harvard.edu/abs/2020A%26A...644A.132U/abstract))!

**Comment by fjebaker** [comment]
The rule of thumb I'd been adhering to thus far is that the more GR effects you add, the more dependant on the metric the result becomes, and so therefore the more important deviations effects should be. But it's entirely non-linear, and sometimes the effects even seem to cancel themselves out -- modelling more of the physics may actually add degeneracies back in, as I believe is the case when looking at certain lineprofiles with self-consistent emissivities.
